crates/core/Cargo.toml

```diff
@@ -72,14 +114,18 @@ rocksdb = { workspace = true, optional = true }
 tikv = { workspace = true, optional = true }
 
 # Other crates
+path-clean.workspace = true
```

Comment:
Do we really need this dependency? The functional part of the library is around 20 lines of code. Lets avoid a left-pad if we can.

Reply:
It is used. `SURREAL_FILE_ALLOWLIST` previously used fs::canonicalize, which does the path cleaning plus it filtered it out if the path did not exist (which we do not want). It now uses the trait from path-clean to do that, but to only strip the path.

Reply:
Even so, I don't think we should use it as a dependency, the twenty lines of code that it provides is not worth the increased exposure to supply chain issues.

Reply:
It's already used within surrealdb itself aswell (so already in supply chain), downloaded millions of times. I don't really see the problem, honestly